### PR TITLE
enhance the mount type check, to allow zfs file systems, where the

### PR DESCRIPTION
**Pull request by achims311**
filesystem name does not exist and does not start with a "/" Fixes Issue #7

### CODE REVIEW

File: fatrace.c [hunk 1]
```diff
@@ -375,9 +375,12 @@ setup_fanotify (int fan_fd)
...
          * point. The others are stuff like proc, sysfs, binfmt_misc etc. which
-         * are virtual and do not actually cause disk access. */
-        if (mount->mnt_fsname == NULL || access (mount->mnt_fsname, F_OK) != 0 ||
-            mount->mnt_fsname[0] != '/') {
+         * are virtual and do not actually cause disk access. 
+         * zfs mount point don't start with a "/" so allow them anyway */
+        if (mount->mnt_fsname == NULL || 
+            (strcmp(mount->mnt_type, "zfs") != 0 && (access (mount->mnt_fsname, F_OK) != 0 || 
+                                                    mount->mnt_fsname[0] != '/'))
+            ) {
```
Comment (martinpitt) [inline fatrace.c]: I stared at this for a bit, and logically it's correct. However, I find it hard to read. It's also indented wrongly (off by one space).

I suggest to leave the original condition as-is, and instead check for zfs inside. Like this:

```C
if (original || condition) {
    /* zfs mount point don't start with a "/" so allow them anyway */
   if (strcmp(mount->mnt_type, "zfs") != 0) {
        debug();
        continue;
   }
}
```
WDYT?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thank you for fixing this!

**Comment by achims311** [comment]
I changed it the way you suggested.
Yes, looks cleaner.

**Review by martinpitt — APPROVED** [review]
Thanks! This looks nicer. Note that I'll squash your commits together to retain a clean history. Don't be afraid of force-pushing, BTW! GitHub handles that just fine and even generates nice inter-diffs between the pushes.

**Comment by martinpitt** [comment]
I did a new release to pick this up, it was about time anyway: https://github.com/martinpitt/fatrace/releases/tag/0.17.0

**Pull request merged**